subsys/audio_module/audio_module.h

```diff
@@ -0,0 +1,465 @@
+	 */
+	int (*data_process)(struct audio_module_handle_private *handle,
+			    struct audio_data const *const audio_data_rx,
+			    struct audio_data *audio_data_tx);
+};
+
+/**
+ * @brief A module's minimum description.
+ */
+struct audio_module_description {
+	/* The module base name. */
+	char *name;
+
+	/* The module type. */
+	enum audio_module_type type;
+
+	/* A pointer to the functions in the module. */
+	const struct audio_module_functions *functions;
+};
+
+/**
+ * @brief Module's thread configuration structure.
+ */
+struct audio_module_thread_configuration {
+	/* Thread stack. */
+	k_thread_stack_t *stack;
+
+	/* Thread stack size. */
+	size_t stack_size;
+
+	/* Thread priority. */
+	int priority;
+
+	/* A pointer to a module's audio data receiver FIFO, can be NULL. */
+	struct data_fifo *msg_rx;
+
+	/* A pointer to a module's audio data transmitter FIFO, can be NULL. */
+	struct data_fifo *msg_tx;
+
+	/* A pointer to the audio data buffer slab, can be NULL. */
+	struct k_mem_slab *data_slab;
+
+	/* Size of each memory data buffer in bytes that will be
+	 * taken from the audio data buffer slab or 0.
+	 */
+	size_t data_size;
+};
+
+/**
+ * @brief Module's generic set-up structure.
+ */
+struct audio_module_parameters {
+	/* The module's description. */
+	struct audio_module_description *description;
+
+	/* The module's thread setting. */
+	struct audio_module_thread_configuration thread;
+};
+
+/**
+ * @brief Private module handle.
+ */
+struct audio_module_handle {
+	/* The unique name of this module instance. */
+	char name[CONFIG_AUDIO_MODULE_NAME_SIZE];
+
+	/* The module's description. */
+	const struct audio_module_description *description;
+
+	/* Current state of the module. */
+	enum audio_module_state state;
+
+	/* Thread ID. */
+	k_tid_t thread_id;
+
+	/* Thread data. */
+	struct k_thread thread_data;
+
+	/* Flag to indicate if the module should put it's output audio data onto it's TX FIFO. */
+	bool use_tx_queue;
+
+	/* List node (pointer to next audio module). */
+	sys_snode_t node;
+
+	/* A single linked-list of the handles the module is connected to. */
+	sys_slist_t hdl_dest_list;
+
+	/* Number of destination modules. */
+	uint8_t dest_count;
+
+	/* Semaphore to count messages between modules. */
+	struct k_sem sem;
+
+	/* Mutex to make the above destinations list thread safe. */
+	struct k_mutex dest_mutex;
+
+	/* Module's thread configuration. */
+	struct audio_module_thread_configuration thread;
+
+	/* Private context for the module. */
+	struct audio_module_context *context;
+};
+
+/**
+ * @brief Private structure describing a data_in message into the module thread.
+ */
+struct audio_module_message {
+	/* Audio data to input. */
+	struct audio_data audio_data;
+
+	/* Sending module's handle. */
+	struct audio_module_handle *tx_handle;
+
+	/* Callback for when the audio data has been consumed. */
+	audio_module_response_cb response_cb;
+};
+
+/**
+ * @brief Function for opening a module.
```

Comment:
```suggestion
 * @brief Open an audio module.
```
Same for the other function @brief in this file that start with "Function"